driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -158,24 +158,51 @@ public Connection addConnectionToClose(int shardId, Connection c) {
     }
   }
 
-  private boolean canUseAdvancedShardAwareness() {
+  private boolean canUseAdvancedShardAwareness(boolean logResult) {
     ShardingInfo shardingInfo = host.getShardingInfo();
     if (shardingInfo == null) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because sharding info is missing");
+      }
       return false;
     }
     if (!manager.configuration().getProtocolOptions().isUseAdvancedShardAwareness()) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because it's disabled in configuration");
+      }
       return false;
     }
 
     boolean isSSLUsed = null != manager.configuration().getProtocolOptions().getSSLOptions();
     if (shardingInfo.getShardAwarePort(isSSLUsed) == 0) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because it's impossible to get the outgoing port");
```

Comment:
I think this message may be wrong. The check isn't about allocating local, outgoing port (it happens elsewhere), it's about whether the server sent us it's shard-aware port number.

Reply:
Fair point. Thanks. Fixed.